test/types/message.spec.js

```diff
@@ -25,19 +25,14 @@ describe('BitswapMessage', () => {
   })
 
   describe('.addEntry', () => {
-    it('want type defaults to want block', () => {
+    it('want type defaults to want block', async () => {
       const cid = cids[1]
       const msg = new BitswapMessage(true)
       msg.addEntry(cid, 1)
       const serialized = msg.serializeToBitswap100()
 
-      expect(Message.decode(serialized).wantlist.entries[0]).to.be.eql({
-        block: cid.bytes,
-        priority: 1,
-        cancel: false,
-        sendDontHave: false,
-        wantType: Message.Wantlist.WantType.Block
-      })
+      const deserialized = await BitswapMessage.deserialize(serialized)
+      expect(deserialized.wantlist.get(cid.toString())).to.have.nested.property('entry.wantType', Message.Wantlist.WantType.Block)
```

Comment:
Protons did some non-standard things here - sendDontHave and wantType should come back as undefined but instead came back as `false` and `0` respectively in line with the comment in the .proto definition.

Protobufjs follows the spec and does not do this so we can't test against the output of the protobuf deserializer but neither should we, really as it's not the unit under test.

Reply:
protons was a mess, happy that we are getting rid of it